### PR TITLE
Update RTProtocol: Static Char Arrays --> std::string Objects [QTM 2024.3 Release]

### PR DESCRIPTION
**Pull request by OliverGlandberger**
**Blocked: Waiting for 2024.3 Release before this can be merged.**

Some RTProtocol files were recently refactored in QTM (see recently-completed [Refactor / Bug Fix] PR for more info). This PR implements the same updates to the same files.

### CODE REVIEW

File: Qualisys/QTMConnect/Source/ThirdParty/RTClientSDK/Include/RTProtocol.h [hunk 1]
```diff
@@ -690,17 +690,17 @@ class DLL_EXPORT CRTProtocol
...
     bool       Reprocess();
 
     static bool GetEventString(CRTPacket::EEvent eEvent, char* pStr);
     static bool ConvertRateString(const char* pRate, EStreamRate &eRate, unsigned int &nRateArg);
-    static unsigned int ConvertComponentString(const char* pComponentType);
-    static bool GetComponentString(char* pComponentStr, unsigned int nComponentType, const SComponentOptions& options = SComponentOptions());
-    static std::vector<std::pair<unsigned int, std::string>> GetComponents(const std::string componentsString);
-    static bool GetComponent(std::string componentStr, unsigned int &component, std::string &option);
+    static unsigned int ConvertComponentString(const std::string& componentsString);
+    static bool GetComponentString(std::string& pComponentStr, unsigned int nComponentType, const SComponentOptions& options = SComponentOptions());
```
Comment (OliverGlandberger) [inline Qualisys/QTMConnect/Source/ThirdParty/RTClientSDK/Include/RTProtocol.h]: **pComponentStr** --> **componentStr**

File: Qualisys/QTMConnectLiveLink/Source/ThirdParty/RTClientSDK/Include/RTProtocol.h [hunk 2]
```diff
@@ -690,17 +690,17 @@ class DLL_EXPORT CRTProtocol
...
     bool       Reprocess();
 
     static bool GetEventString(CRTPacket::EEvent eEvent, char* pStr);
     static bool ConvertRateString(const char* pRate, EStreamRate &eRate, unsigned int &nRateArg);
-    static unsigned int ConvertComponentString(const char* pComponentType);
-    static bool GetComponentString(char* pComponentStr, unsigned int nComponentType, const SComponentOptions& options = SComponentOptions());
-    static std::vector<std::pair<unsigned int, std::string>> GetComponents(const std::string componentsString);
-    static bool GetComponent(std::string componentStr, unsigned int &component, std::string &option);
+    static unsigned int ConvertComponentString(const std::string& componentsString);
+    static bool GetComponentString(std::string& pComponentStr, unsigned int nComponentType, const SComponentOptions& options = SComponentOptions());
```
Comment (OliverGlandberger) [inline Qualisys/QTMConnectLiveLink/Source/ThirdParty/RTClientSDK/Include/RTProtocol.h]: pComponentStr --> componentStr

File: Qualisys/QTMConnect/Source/ThirdParty/RTClientSDK/Include/RTProtocol.h (hunk 1 above)
Comment (OliverGlandberger) [reply Qualisys/QTMConnect/Source/ThirdParty/RTClientSDK/Include/RTProtocol.h]: Updated.

File: Qualisys/QTMConnectLiveLink/Source/ThirdParty/RTClientSDK/Include/RTProtocol.h (hunk 2 above)
Comment (OliverGlandberger) [reply Qualisys/QTMConnectLiveLink/Source/ThirdParty/RTClientSDK/Include/RTProtocol.h]: Updated.

**Review by MrCeliaki — APPROVED** [review]

**Review by MrCeliaki — APPROVED** [review]

**Pull request merged**